File: event/models/tx.js
'use strict';
module.exports = (sequelize, DataTypes) => {
  var Tx = sequelize.define('Tx', {
    id: {
      type: DataTypes.INTEGER,
      autoIncrement: true,
      primaryKey: true,
      allowNull: false,
      unique: true,
    },
    hash: DataTypes.STRING,
    scope: DataTypes.STRING,
    contract_name: DataTypes.STRING,
    contract_address: DataTypes.STRING,
    contract_method: DataTypes.STRING,
    arguments: DataTypes.STRING,
    payload: DataTypes.TEXT,
    from_address: DataTypes.STRING,
    to_address: DataTypes.STRING,
    amount: DataTypes.STRING,
    user_id: DataTypes.INTEGER,
    status: DataTypes.INTEGER,
    transaction_status: DataTypes.STRING,
    chain_id: DataTypes.INTEGER
  }, {
    tableName: 'tx',
    timestamps: false,
    underscored: true,
  });

  return Tx;
};

File: event/models/tx.go
package models

type Tx struct {
	Hash            string `gorm:"column:hash;" json:"hash"`
	ContractAddress string `gorm:"column:contract_address;" json:"contract_address"`
	ContractMethod  string `gorm:"column:contract_method;" json:"contract_method"`
	Payload         string `gorm:"column:payload;" json:"payload"`
	Status          int    `gorm:"column:status;default:-1;" json:"status"`
	ChainID         int    `gorm:"column:chain_id;default:4;" json:"chain_id"`
	Offchain        string `gorm:"column:offchain;" json:"offchain"`
}

func (u Tx) TableName() string {
	return "tx"
}
